src/NuGet.Core/NuGet.Common/RuntimeEnvironmentHelper.cs

```diff
@@ -36,7 +41,37 @@ public static bool IsWindows
 
         public static bool IsMono
         {
-            get { return _isMono.Value; }
+            get
+            {
+                if (IsRunningInVisualStudio)
+                {
+                    // skip Mono type check if current process is Devenv
+                    return false;
+                }
+
+                return _isMono.Value;
+            }
+        }
+
+        public static bool IsRunningInVisualStudio
+        {
+            get
+            {
+                var currentProcessPath = GetCurrentProcessFilePath();
```

Comment:
You'll probably want to make this lazy and only do the check once too right?

Reply:
Making it lazy might not make a diff here since it will always be evaluated once IsMono is called. Although checking it once makes sense. I'll change that